crates/fluvio-controlplane-metadata/src/mirroring/mod.rs

```diff
@@ -0,0 +1,118 @@
+use fluvio_types::ReplicaMap;
+
+use fluvio_protocol::{Decoder, Encoder};
+use fluvio_protocol::api::Request;
+use fluvio_protocol::link::ErrorCode;
+
+use crate::topic::TopicSpec;
+
+#[derive(Encoder, Decoder, Default, Debug, Clone)]
+pub struct MirroringRemoteClusterRequest<S> {
+    pub request: S,
+}
+
+impl<S> MirroringRemoteClusterSpec for MirroringRemoteClusterRequest<S> where S: Encoder + Decoder {}
+
+#[derive(Encoder, Decoder, Default, Debug, Clone, PartialEq)]
+#[cfg_attr(
+    feature = "use_serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct MirrorConnect {
+    pub name: String,
+}
+
+impl MirroringRemoteClusterSpec for MirrorConnect {}
+
+impl Spec for MirrorConnect {
+    const LABEL: &'static str = "MirroringConnect";
+    type IndexKey = String;
+    type Status = MirroringStatus;
+    type Owner = Self;
+}
+
+#[derive(Encoder, Decoder, Default, Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "use_serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct MirroringStatus {}
+
+impl std::fmt::Display for MirroringStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "MirroringStatus")
+    }
+}
+
+impl Status for MirroringStatus {}
+
+impl Request for MirroringRemoteClusterRequest<MirrorConnect> {
+    const API_KEY: u16 = MirroringApiKey::Connect as u16;
+    type Response = MirroringStatusResponse;
+}
+
+/// API call from client to SPU
+#[repr(u16)]
+#[derive(Encoder, Decoder, Eq, PartialEq, Debug, Clone, Copy)]
+#[cfg_attr(feature = "use_serde", derive(serde::Serialize, serde::Deserialize))]
+#[fluvio(encode_discriminant)]
+pub enum MirroringApiKey {
+    ApiVersion = 1, // version api key
+    Connect = 3000,
+}
+
+impl Default for MirroringApiKey {
+    fn default() -> Self {
+        Self::ApiVersion
+    }
+}
+
+pub trait MirroringRemoteClusterSpec: Encoder + Decoder {}
+
+#[derive(Encoder, Decoder, Default, Debug, Clone, PartialEq)]
+#[cfg_attr(
+    feature = "use_serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct MirroringStatusResponse {
+    pub name: String,
+    #[cfg_attr(feature = "use_serde", serde(skip))]
+    pub error_code: ErrorCode,
+    pub error_message: Option<String>,
+    pub topics: Vec<MirroringSpecWrapper<TopicSpec>>,
+}
+
+impl MirroringStatusResponse {
+    pub fn new_ok(name: &str, topics: Vec<MirroringSpecWrapper<TopicSpec>>) -> Self {
+        MirroringStatusResponse {
+            name: name.to_string(),
+            error_code: ErrorCode::None,
+            error_message: None,
+            topics,
+        }
+    }
+}
+
+#[derive(Encoder, Decoder, Default, Debug, Clone, PartialEq)]
+#[cfg_attr(
+    feature = "use_serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(rename_all = "camelCase")
+)]
+pub struct MirroringSpecWrapper<S> {
+    pub key: String,
+    pub spec: S,
+    pub replica_map: ReplicaMap,
```

Comment:
Why replica map is needed?

Reply:
To have the spu_id, I think that we will need it on spu-to-spu part.